src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicDetectPreviewFeatureAnalyzer.vb

```diff
@@ -11,23 +12,262 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
+                Next
+            Next
+
+            previewInterfaceNode = Nothing
+            Return False
+        End Function
+
         Protected Overrides Function GetPreviewInterfaceNodeForTypeImplementingPreviewInterface(typeSymbol As ISymbol, previewInterfaceSymbol As ISymbol) As SyntaxNode
+            Dim typeSymbolDeclaringReferences = typeSymbol.DeclaringSyntaxReferences
+
+            For Each syntaxReference In typeSymbolDeclaringReferences
+                Dim typeSymbolDefinition = syntaxReference.GetSyntax()
+                Dim classStatement = TryCast(typeSymbolDefinition, ClassStatementSyntax)
+                If classStatement IsNot Nothing Then
+                    Dim classBlock = TryCast(classStatement.Parent, ClassBlockSyntax)
+                    If classBlock IsNot Nothing Then
+                        Dim classImplement = classBlock.Implements
+                        Dim syntaxNode As SyntaxNode = Nothing
+                        If TryGetPreviewInterfaceNodeForClassOrStructImplementingPreviewInterface(classImplement, previewInterfaceSymbol, syntaxNode) Then
+                            Return syntaxNode
+                        End If
+                    End If
+                End If
+
+                Dim structStatement = TryCast(typeSymbolDefinition, StructureStatementSyntax)
+                If structStatement IsNot Nothing Then
+                    Dim structBlock = TryCast(structStatement.Parent, StructureBlockSyntax)
+                    If structBlock IsNot Nothing Then
+                        Dim structImplement = structBlock.Implements
+                        Dim syntaxNode As SyntaxNode = Nothing
+                        If TryGetPreviewInterfaceNodeForClassOrStructImplementingPreviewInterface(structImplement, previewInterfaceSymbol, syntaxNode) Then
+                            Return syntaxNode
+                        End If
+                    End If
+                End If
+
+                Dim interfaceStatement = TryCast(typeSymbolDefinition, InterfaceStatementSyntax)
+                If interfaceStatement IsNot Nothing Then
+                    Dim interfaceBlock = TryCast(interfaceStatement.Parent, InterfaceBlockSyntax)
+                    If interfaceBlock IsNot Nothing Then
+                        Dim interfaceImplement = interfaceBlock.Inherits
+                        Dim syntaxNode As SyntaxNode = Nothing
+                        If TryGetPreviewInterfaceNodeForClassOrStructImplementingPreviewInterface(interfaceImplement, previewInterfaceSymbol, syntaxNode) Then
+                            Return syntaxNode
+                        End If
+                    End If
+                End If
+            Next
+
             Return Nothing
         End Function
 
         Protected Overrides Function GetConstraintSyntaxNodeForTypeConstrainedByPreviewTypes(typeOrMethodSymbol As ISymbol, previewInterfaceConstraintSymbol As ISymbol) As SyntaxNode
             Return Nothing
```

Comment:
I have tried to implement this but could not find a way to get implementation constraints from `ClassStatementSyntax` in VB, [filed an issue](https://github.com/dotnet/roslyn-analyzers/issues/5646) for that.

Reply:
Ah, I suspect this is the reason why `AFoo` was highlighted above instead of `Foo`!